### PR TITLE
[SYCL] Fix specialization constants struct members

### PR DESCRIPTION
**Pull request by Fznamznon**
FE crashed on attempt to create initializer for struct with spec
constant members because there was no initializers for spec const
fields. Added default initialization for spec constants.

### CODE REVIEW

File: sycl/include/CL/sycl/experimental/spec_constant.hpp [hunk 1]
```diff
@@ -32,7 +32,9 @@ template <typename T, typename ID = T> class spec_constant {
 private:
   // Implementation defined constructor.
 #ifdef __SYCL_DEVICE_ONLY__
+public:
   spec_constant() {}
+private:
```
Comment (s-kanaev) [inline sycl/include/CL/sycl/experimental/spec_constant.hpp:37]: This change will violate the spec, won't it?

Comment (kbobrovs) [reply sycl/include/CL/sycl/experimental/spec_constant.hpp:37]: I don't see how this violates the [spec](https://github.com/codeplaysoftware/standards-proposals/blob/master/spec-constant/index.md). Plus I do think that given the experimental and temporary status of the implemented spec we have certain degree of freedom to adjust it if we find it reasonable.

**Comment by Fznamznon** [comment]
@s-kanaev , @kbobrovs , could you please give a formal approve? There was no functional changes in spec const part and runtime test.

**Comment by s-kanaev** [comment]
@Fznamznon  Sorry, I've missed the notification somehow.

**Pull request merged**